src/System.Slices/System/Buffers/IOutput.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Buffers
+{
+    public interface IOutput
+    {
+        Span<byte> Buffer { get; }
+        void Advance(int bytes);
+
+        /// <summary>desiredBufferLength == 0 means "i don't care"</summary>
+        void Enlarge(int desiredBufferLength = 0);
```

Comment:
If you have enough, don't call Enlarge, i.e.
if(out.Buffer.Length < needed) out.Enlarge(needed);
out.Buffer.Set(data);